aiidalab_widgets_base/nodes.py

```diff
@@ -141,6 +141,15 @@ def _refresh_output(self):
             display(self._tree)
 
     def _observe_tree_selected_nodes(self, change):
+        for node in change["new"]:
+            # find the selected node and build the tree from it, so that users can expand and explore the tree
+            node_pk = (
+                node.parent_pk
+                if isinstance(node, AiidaOutputsTreeNode)
+                else getattr(node, "pk", None)
+            )
+
+            self._build_tree(self.find_node(node_pk, getattr(node, "namespaces", None)))
```

Comment:
From just looking at this code, I wonder how is it ensured that the tree is build only once? i.e. if a user selects / deselects the same node multiple times, is it going to trigger the build_tree multiple times or will the existing nodes be reused somehow?

Reply:
Thanks for the review. It's designed to build the tree every time the user selects it, for two reasons:
- if the job is running, it will ensure the user sees the latest node tree.
- since it only builds the first level of the child nodes, it should be quick even for a large number of child nodes.

Reply:
Hi @danielhollas , the already loaded child nodes will be reused when running the `_build_tree`.